src/cdk-pipeline-migrate.ts

```diff
@@ -0,0 +1,163 @@
+import { BuildSpec, BuildEnvironmentVariableType, LinuxBuildImage, PipelineProject, PipelineProjectProps } from '@aws-cdk/aws-codebuild'
+import { Artifact } from '@aws-cdk/aws-codepipeline'
+import { SecurityGroup } from '@aws-cdk/aws-ec2'
+import { Secret } from '@aws-cdk/aws-secretsmanager'
+import { StringParameter } from '@aws-cdk/aws-ssm'
+import { CodeBuildAction } from '@aws-cdk/aws-codepipeline-actions'
+import { PolicyStatement } from '@aws-cdk/aws-iam'
+import { Construct, Fn } from '@aws-cdk/core'
+import { FoundationStack } from './foundation-stack'
+
+export interface ICDKPipelineDeployProps extends PipelineProjectProps {
+
+    /**
+     * Application source artifact.
+     */
+    readonly appSourceArtifact: Artifact;
+
+    /**
+     * Namespace to use for stack names etc
+     */
+    readonly namespace: string;
+
+    readonly contextEnvName: string;
+    readonly appBuildCommands?: string[];
+    readonly postDeployCommands?: string[];
+    /**
+     * The SSM prefix to use for any environment variables
+     */
+    readonly ssmPrefix: string
+    /**
+     * The path to the Secrets Manager secret to allow authenticated Docker logins
+     */
+    readonly dockerCredentialsPath: string
+
+    /**
+     * Any runtime environments needed in addition to the one needed for cdk itself (currently nodejs: '12.x')  e.g. `python: '3.8'`
+     */
+    readonly additionalRuntimeEnvironments?: { [key: string]: string };
+    /**
+     * A Foundation Stack that contains VPC information
+     */
+    readonly foundationStack: FoundationStack
+  }
+
+export class CDKPipelineMigrate extends Construct {
```

Comment:
This isn't really a CDK thing. It's more a rails migration type action.
```suggestion
export class RailsMigration extends Construct {
```

I'm also still not convinced this is a good way to handle this, or that it will work for other pipelines without having to override nearly everything. I'd prefer we just reuse the image built from the previous stage, either by pulling it from whatever ECR/Tag was pushed by cdk in the previous stage, or by running an ECS task with command overrides using the task definition deployed in the previous stage.